### ISSUE TITLE
Getting a TypeError from bootimage.py, process fails

### ISSUE THREAD

**Issue by oddgirl** [chenxiaolong/avbroot#50]
Hi! 
The total output looks like : 

`
  File "/home/user/androidstuff/avbroot/avbroot.py", line 3, in <module>
    from avbroot import main
  File "/home/user/androidstuff/avbroot/avbroot/main.py", line 13, in <module>
    from . import boot
  File "/home/user/androidstuff/avbroot/avbroot/boot.py", line 12, in <module>
    from .formats import bootimage
  File "/home/user/androidstuff/avbroot/avbroot/formats/bootimage.py", line 115, in <module>
    class BootImage:
  File "/home/user/androidstuff/avbroot/avbroot/formats/bootimage.py", line 118, in BootImage
    f: None | typing.BinaryIO = None,
TypeError: unsupported operand type(s) for |: 'NoneType' and 'type'

`

Is this something broken on my end (python 3.9.2) or not?

**Comment by pascallj** [comment]
The UnionType (`x | y`) is indeed [added in Python 3.10](https://docs.python.org/3/library/types.html#types.UnionType).

By the exact version of Python you are using (3.9.2), I assume you are running Debian 11. If so, if you want you can try my [backport of Python 3.10 for Debian](https://github.com/pascallj/python3.10-backport). You can then create a virtual environment like this:

```sh
mkdir avbroot_venv
python3.10 -m venv avbroot_venv
cd avbroot_venv
source bin/activate
```

Install `lz4` and `protobuf` while activated your virtual environment:

```sh
pip install lz4 protobuf
```

Then you should be able to run avbroot with Python 3.10. I haven't tested this though. As always, be careful when installing software from a third party.

@chenxiaolong, do you intend to set a minimum Python version for this project?

**Comment by chenxiaolong** [comment]
> @chenxiaolong, do you intend to set a minimum Python version for this project?

Yeah, I'll document that in the README. Right now, it's de-facto Python 3.10 due to the distros I test on (Fedora 37, Arch, Ubuntu 22.04/22.10, Alpine 3.17).

> The UnionType (`x | y`) is indeed [added in Python 3.10](https://docs.python.org/3/library/types.html#types.UnionType).

If this is the only thing blocking 3.9 compatibility, I'll change it. (and maybe add Debian 11 to the list of distros I test on)

**Comment by chenxiaolong** [comment]
This is fixed in #52. I've also added Debian 11 in the new testing framework (#51) so hopefully these compatibility issues won't pop up anymore.
